schema_editor/test/spec/schemas/schemas-service.spec.js

```diff
@@ -148,4 +176,69 @@ describe('ase.schemas:Schemas', function () {
             message: jasmine.stringMatching('Diff')
         }]));
     });
+
+    // NOTE: This test must be updated when new field types are added.
+    it('should preserve all information when serializing and deserializing schemas', function () {
+        // In practice this means that the serialize and deserialize methods are inverses
```

Comment:
Would it be sensible to somehow test that all of the fieldTypes are represented below, and if not automatically fail the test?